### DISCUSSION TITLE
请问大佬们，tir层面的粒度是不支持算子库映射的吧？

### DISCUSSION THREAD

**Discussion by eAzure** [mlc-ai/mlc-zh#21] [Q&A] [answered]
各位大佬们好，看到最新一课中用到的MyModule（实现了向量加法），感觉tir这个粒度已经到了for这种，是不已无法实现向算子库的映射（如cudnn）？想问一下，TVM现在支持cudnn是在哪个层面实现的呢（relay层吗）？

**Comment by Hzfengsy** [comment, marked as answer]
TVM支持cudnn是在图层（relay/relax），可以通过BYOC来实现

**Reply by eAzure** [reply]
大佬，如果网络一开始用的BYOC这种方式接入了cudnn，对于cudnn中不支持的算子，之后是走的tir这条路吗？即生成cuda c这种或llvm相关的吗？谢谢！
